service/jmx/src/main/java/org/jolokia/service/jmx/handler/WriteHandler.java

```diff
@@ -132,21 +133,15 @@ private Object setAttribute(JolokiaWriteRequest request, MBeanServerConnection s
 
             // We set an inner value, hence we have to return provided value itself for resetting
             // it later back via JMX
-            return new Object[] {
-                    pCurrentValue,
-                    context.getMandatoryService(Serializer.class).setInnerValue(pCurrentValue, newValue, pathParts)
-            };
+            return context.getMandatoryService(Serializer.class).setInnerValue(pCurrentValue, newValue, pathParts);
```

Comment:
Sorry, I still didn't get it ;-) The first argument to `setInnerValue` is the object on which to set a property, either a direct property or one which is nested deeper when a path is used. This object, which you provide as argument, is returned _unchanged_ as part of the return value (in `WriteRequestValue`). 

At least this is the case for the [default serializer](https://github.com/rhuss/jolokia/blob/2.0/service/serializer/src/main/java/org/jolokia/service/serializer/JolokiaSerializer.java#L77-L79). If you can convince me that:

* there are situations where `WriteRequestValue.oldValue` is different from `pCurentValue` (the first argument to `setInnerValue()`
* how it helps in diving into the object when a path is given (e.g. to avoid reflection)

then I'm all for it. Otherwise its only boilerplate, if the value given as argument is always returned because a called can easily remember the argument to work with it after the call to the serializer.

 But I guess I'm just a bit lamebrained ;-)